FastqError2.py

```diff
@@ -0,0 +1,155 @@
+    for line in infile:
+        line = line.rstrip()
+        if line[0] == '@':
+            #Creates a key in the dictionary of the name of the sequence and sets the value to the sequence
+            sequences[line[1:].rstrip()] = next(infile).rstrip()
+    return sequences
+
+def parse_cmdline_params(arg_list=None):
+    """Parses command line arguments.
+    :param arg_list: arguments from the command line
+    :type arg_list: list
+    :return: dictionary of options
+    """
+
+    description = "Calculate the GC percentage of sequences in a fasta file"
+    parser = ArgumentParser(description=description,
+                            formatter_class=ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument("--fastq", "-f", help="FASTQ file to process", required=True, type=FileType('r'))
+
+    # Parse options
+    opts = parser.parse_args(args=arg_list)
+    return opts
+
+def createQualityScores(sequence):
+    """Makes a string of ascii characters corresponding to different quality values.
+
+    :param str sequence: sequence to be translated
+    :return: str new_ascii: ascii characters corresponding to the quality values determined by the error probability
+    """
+    ascii_ref = "!\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~"
+    position = 0
+    new_ascii = ''
+    for base in sequence:
+        position += 1
+        error_prob = position * 0.001
+        quality_score = -10 * math.log10(error_prob)
+        new_ascii += ascii_ref[int(quality_score)]
+    return new_ascii
+
+def main(args):
+    opts = parse_cmdline_params(args[1:])
+    sequences = createFastqDict(opts.fastq)
+    #sequences = createFastqDict('sampleFastq')
+    total_error_num = 0
+    total_errors = 0
+    total_lengths = 0
+    for key in sequences:
+        error_seq_string = ''
+        seq_name = key
+        seq_string = sequences[key]
+        position = 0
+        error_num = 0
+        quality_string = createQualityScores(seq_string)
+        for base in seq_string:
+            if base not in 'AGCT':
+                raise ValueError("This base,", base ,",is not an A, G, C, or T!")
+            position += 1
+            error_prob = 0.001 * position
+            if error_prob * 100 <= 0:
+                raise ValueError("The error probability must be greater than 0!")
+            #Checks if it needs to add an error
+            if (random.randint(1,1000) < (error_prob)*1000):
+                error_seq_string = addError(error_seq_string,base,True)
+                error_num += 1
+            else:
+                error_seq_string = addError(error_seq_string,base,False)
+        total_errors += error_num
+        total_error_num += 1
+        error_perc = error_num/len(seq_string)
+        total_lengths += len(seq_string)
+        printErrors(blue(seq_name),seq_string,error_seq_string,green(str(error_num)))
+        print 'The error percent for this sequence is:',yellow(str(error_perc*100)),'%'
```

Comment:
Print these to stderr instead of stdout. Only the fastq should be written to stdout. This allows you to pipe the output of this file into any other program that handles fastqs.